### ISSUE TITLE
How to build multilabel text classfication dataset

### ISSUE THREAD

**Issue by HenryYuen128** [huggingface/setfit#191]
From the sample below, param **column_mapping** is used to set up the dataset. What is the format of label column in multilabel?Is it the one-hot label?

trainer = SetFitTrainer(
    model=model,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss_class=CosineSimilarityLoss,
    metric="accuracy",
    batch_size=16,
    num_iterations=20, # The number of text pairs to generate for contrastive learning
    num_epochs=1, # The number of epochs to use for constrastive learning
    column_mapping={"sentence": "text", "label": "label"} # Map dataset columns to text/label expected by trainer

**Comment by tomaarsen** [comment]
Please have a look at the [text-classification_multilabel.ipynb](https://github.com/huggingface/setfit/blob/main/notebooks/text-classification_multilabel.ipynb) example notebook for information on how to do multilabel training. 

The `train_dataset` used in that notebook is:
```python
Dataset({
    features: ['text', 'violence', 'directed_vs_generalized', 'gender', 'race', 'national_origin', 'disability', 'religion', 'sexual_orientation', 'labels'],
    num_rows: 64
})
```
from the ethos multilabel dataset.

In this dataset, `train_dataset["labels"]` looks like so:
```python
[[1, 1, 0, 0, 0, 0, 0, 1],
 [1, 1, 0, 0, 0, 0, 1, 0],
 [1, 0, 0, 0, 1, 0, 0, 0],
 [1, 0, 0, 1, 0, 0, 0, 0],
 [1, 1, 0, 0, 1, 0, 0, 0],
...
```
Note that there are 8 features, hence the length of these lists.

This is then used like
```python
model = SetFitModel.from_pretrained(model_id, multi_target_strategy="one-vs-rest")

trainer = SetFitTrainer(
    model=model,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss_class=CosineSimilarityLoss,
    num_iterations=20,
    column_mapping={"text": "text", "labels": "label"},
)
```

- Tom Aarsen

**Comment by HenryYuen128** [comment]
> Please have a look at the [text-classification_multilabel.ipynb](https://github.com/huggingface/setfit/blob/main/notebooks/text-classification_multilabel.ipynb) example notebook for information on how to do multilabel training.
> 
> The `train_dataset` used in that notebook is:
> 
> ```python
> Dataset({
>     features: ['text', 'violence', 'directed_vs_generalized', 'gender', 'race', 'national_origin', 'disability', 'religion', 'sexual_orientation', 'labels'],
>     num_rows: 64
> })
> ```
> 
> from the ethos multilabel dataset.
> 
> In this dataset, `train_dataset["labels"]` looks like so:
> 
> ```python
> [[1, 1, 0, 0, 0, 0, 0, 1],
>  [1, 1, 0, 0, 0, 0, 1, 0],
>  [1, 0, 0, 0, 1, 0, 0, 0],
>  [1, 0, 0, 1, 0, 0, 0, 0],
>  [1, 1, 0, 0, 1, 0, 0, 0],
> ...
> ```
> 
> Note that there are 8 features, hence the length of these lists.
> 
> This is then used like
> 
> ```python
> model = SetFitModel.from_pretrained(model_id, multi_target_strategy="one-vs-rest")
> 
> trainer = SetFitTrainer(
>     model=model,
>     train_dataset=train_dataset,
>     eval_dataset=eval_dataset,
>     loss_class=CosineSimilarityLoss,
>     num_iterations=20,
>     column_mapping={"text": "text", "labels": "label"},
> )
> ```
> 
> * Tom Aarsen

Thanks. And is it possible to apply TrainingArguments in HuggingFace?  It seems that evaluate after each epoch training does not work now?

**Comment by tomaarsen** [comment]
No, the `SetFitTrainer` doesnot accept HuggingFace `TrainingArguments`, sadly. Evaluating after every epoch is only possible by forking this repository and building that in yourself. However, I believe that callbacks for each-epoch evaluation may be implemented in the future.

**Comment by tomaarsen** [comment]
I believe this should be answered now!
